src/OpenLoco/src/World/Company.cpp

```diff
@@ -63,6 +63,21 @@ namespace OpenLoco
         Ui::WindowManager::invalidate(Ui::WindowType::company, enumValue(companyId));
     }
 
+    void Company::clearOwnerStatusForDeletedVehicle(EntityId vehicleId)
+    {
+        // Prevent any possible owner status with a dangling reference to a deleted vehicle
+
+        if (ownerStatus.isEntity() && ownerStatus.getEntity() == vehicleId)
+        {
+            auto previousUpdatingCompanyId = CompanyManager::getUpdatingCompanyId();
```

Comment:
I think we can just do `ownerStatus = OwnerStatus();` no need to make it find a new one just set it back to the null value. Then you don't need to mess about with modifying the update function.

Reply:
Should also check if observationEntity is this id and reset it to null if it is.

Reply:
> I think we can just do `ownerStatus = OwnerStatus();` no need to make it find a new one just set it back to the null value. Then you don't need to mess about with modifying the update function.

That was my initial fix, but found while testing that the state with the window open would be empty and that looked weird.

Reply:
(At least until the next automatic  state update)

Reply:
hmm it's pretty normal for it to be empty. For example if you pick up a vehicle you are watching the viewport goes empty.